xcdat/regridder/regrid2.py

```diff
@@ -46,259 +46,170 @@ def __init__(self, input_grid: xr.Dataset, output_grid: xr.Dataset, **options: A
-        -------
-        Dict
-            Mapping of axis name e.g. ("X", "Y", etc) to output sizes.
-        """
-        output_sizes = {}
 
-        axis_name_map = {y[0]: x for x, y in da.cf.axes.items()}
+def _regrid(
+    input_data_var: xr.DataArray,
+    src_lat_bnds: np.ndarray,
+    src_lon_bnds: np.ndarray,
+    dst_lat_bnds: np.ndarray,
+    dst_lon_bnds: np.ndarray,
+) -> np.ndarray:
+    lat_mapping, lat_weights = _map_latitude(src_lat_bnds, dst_lat_bnds)
+    lon_mapping, lon_weights = _map_longitude(src_lon_bnds, dst_lon_bnds)
 
-        for standard_name in da.sizes.keys():
-            try:
-                axis_name = axis_name_map[standard_name]
-            except KeyError:
-                raise RuntimeError(
-                    f"Could not find axis {standard_name!r}, ensure {standard_name!r} "
-                    "exists and the attributes are correct."
-                )
-
-            if standard_name in self._output_grid:
-                output_sizes[axis_name] = self._output_grid.sizes[standard_name]
-            else:
-                output_sizes[axis_name] = da.sizes[standard_name]
-
-        return output_sizes
-
-    def _regrid(
-        self,
-        input_data: np.ndarray,
-        axis_sizes: Dict[str, int],
-        ordered_axis_names: List[str],
-    ) -> np.ndarray:
-        """Applies regridding to input data.
-
-        Parameters
-        ----------
-        input_data : np.ndarray
-            Input multi-dimensional array on source grid.
-        axis_sizes : Dict[str, int]
-            Mapping of axis name e.g. ("X", "Y", etc) to output sizes.
-        ordered_axis_names : List[str]
-            List of axis name in order of dimensions of ``input_data``.
-
-        Returns
-        -------
-        np.ndarray
-            Multi-dimensional array on destination grid.
-        """
-        input_lat_index = ordered_axis_names.index("Y")
+    # convert to pure numpy
+    input_data = input_data_var.astype(np.float32).data
 
-        input_lon_index = ordered_axis_names.index("X")
+    y_name, y_index = _get_dimension(input_data_var, "Y")
+    x_name, x_index = _get_dimension(input_data_var, "X")
 
-        output_shape = [axis_sizes[x] for x in ordered_axis_names]
+    y_length = len(lat_mapping)
+    x_length = len(lon_mapping)
 
-        output_data = np.zeros(output_shape, dtype=np.float32)
+    other_dims = {
+        x: y for x, y in input_data_var.sizes.items() if x not in (y_name, x_name)
+    }
+    other_sizes = list(other_dims.values())
 
-        base_put_index = self._base_put_indexes(axis_sizes)
+    data_shape = [y_length * x_length] + other_sizes
+    output_data = np.zeros(data_shape, dtype=np.float32)
 
-        for lat_index, lat_map in enumerate(self._lat_mapping):
-            lat_weight = self._lat_weights[lat_index]
+    is_2d = input_data_var.ndim <= 2
 
-            input_lat_segment = np.take(input_data, lat_map, axis=input_lat_index)
+    # need to optimize
```

Comment:
Is this comment still valid? I think you mentioned possibly using Xarray directly here before?

Reply:
Yea this comment is still valid, it's the area I want to continue looking at.